src/com/google/javascript/jscomp/parsing/parser/Scanner.java

```diff
@@ -869,52 +870,18 @@ private static String processUnicodeEscapes(String value) {
     return value;
   }
 
-  @SuppressWarnings("ShortCircuitBoolean") // Intentional to minimize branches in this code
+  /**
+   * Interface from UnicodeRegex. Includes old optimizations.
+   */
   private static boolean isIdentifierStart(char ch) {
-    // Most code is written in pure ASCII, so create a fast path here.
-    if (ch <= 127) {
-      // Intentionally avoiding short circuiting behavior of "||" and "&&".
-      // This minimizes branches in this code which minimizes branch prediction misses.
-      return ((ch >= 'A' & ch <= 'Z') | (ch >= 'a' & ch <= 'z') | (ch == '_' | ch == '$'));
-    }
-
-    // Handle non-ASCII characters.
-    // TODO(tjgq): This should include all characters with the ID_Start property.
-    if (Character.isLetter(ch)) {
-      return true;
-    }
-
-    // Workaround for b/36459436.
-    // When running under GWT/J2CL, Character.isLetter only handles ASCII.
-    // Angular relies heavily on Latin Small Letter Barred O and Greek Capital Letter Delta.
-    // Greek letters are occasionally found in math code.
-    // Latin letters are found in our own tests.
-    return (ch >= 0x00C0 & ch <= 0x00D6) // Latin letters
-        // 0x00D7 = multiplication sign, not a letter
-        | (ch >= 0x00D8 & ch <= 0x00F6) // Latin letters
-        // 0x00F7 = division sign, not a letter
-        | (ch >= 0x00F8 & ch <= 0x00FF) // Latin letters
-        | ch == 0x0275 // Latin Barred O
-        | (ch >= 0x0391 & ch <= 0x03A1) // Greek uppercase letters
-        // 0x03A2 = unassigned
-        | (ch >= 0x03A3 & ch <= 0x03A9) // Remaining Greek uppercase letters
-        | (ch >= 0x03B1 & ch <= 0x03C9); // Greek lowercase letters
-  }
-
-  @SuppressWarnings("ShortCircuitBoolean") // Intentional to minimize branches in this code
+    return UnicodeMatch.isJavascriptIdentifierStart(ch);
```

Comment:
We still want to maintain the "ch <= 127" check. This is a fast path for performance reasons.

Reply:
@concavelenz I retained the fastpath in `UnicodeMatch.java` (line 69):
```java
@SuppressWarnings("ShortCircuitBoolean")
  private static boolean fastPathLetters(char ch) {
    return (
      // ASCII
      ('A' <= ch & ch <= 'Z') |
      ('a' <= ch & ch <= 'z') |
      (ch == '_' | ch == '$') |
      // Latin letters
      (0x00C0 <= ch & ch <= 0x00D6) |
      // Latin letters
      (0x00D8 <= ch & ch <= 0x00F6) |
      // Latin letters
      (0x00F8 <= ch & ch <= 0x00FF) |
      // Latin Barred O
      (ch == 0x0275) |
      // Greek uppercase letters
      (0x0391 <= ch & ch <= 0x03A1) |
      // Remaining Greek uppercase letters
      (0x03A3 <= ch & ch <= 0x03A9) |
      // Greek lowercase letters
      (0x03B1 <= ch & ch <= 0x03C9)
    );
  }

  // Intentional to minimize branches in this code
  @SuppressWarnings("ShortCircuitBoolean")
  private static boolean fastPathDigits(char ch) {
    return '0' <= ch & ch <= '9';
  }

  public static boolean isJavascriptIdentifierStart(char ch) {
    return fastPathLetters(ch)
        || JavascriptIdentifierStart
              .matcher(Character.toString(ch))
              .matches();
  }

  public static boolean isJavascriptIdentifierPart(char ch) {
    return (fastPathLetters(ch) | fastPathDigits(ch))
        || JavascriptIdentifierPart
              .matcher(Character.toString(ch))
              .matches();
  }
```

I tried to retain all existing optimizations.